bhop.s

```diff
@@ -1917,14 +1928,65 @@ cleanup:
 .endproc
 
 .proc play_dpcm_samples
+; if (m_cDAC != 255) {
+    ; WriteRegister(0x4011, m_cDAC);
+    ; m_cDAC = 255;
+; }
+        lda effect_dac_level
+        cmp #$FF
+        beq skip_dac
+        sta $4011
+        lda #$FF
+        sta effect_dac_level
+skip_dac:
+
+; if (mRetriggerPeriod != 0) {
+    ; mRetriggerCtr--;
+    ; if (mRetriggerCtr == 0) {
+        ; mRetriggerCtr = mRetriggerPeriod;
+        ; mEnabled = true;
+        ; mTriggerSample = true;
+    ; }
+; }
+        lda effect_retrigger_period
+        beq next
+        dec effect_retrigger_counter
+        lda effect_retrigger_counter
+        bne next
+        lda effect_retrigger_period
+        sta effect_retrigger_counter
         lda channel_status + DPCM_INDEX
-        and #CHANNEL_SUPPRESSED
-        bne done
-
+        ora #CHANNEL_TRIGGERED
+        and #($FF - CHANNEL_SUPPRESSED)
+        sta channel_status + DPCM_INDEX
+next:
+
+; if (m_bRelease) {
+    ; // Release command
+    ; WriteRegister(0x4015, 0x0F);
+    ; mEnabled = false;
+    ; m_bRelease = false;
+; }
         lda channel_status + DPCM_INDEX
         and #(CHANNEL_MUTED | CHANNEL_RELEASED)
         bne dpcm_muted
 
+; if (!mEnabled)
+    ; return;
+        lda channel_status + DPCM_INDEX
+        and #CHANNEL_SUPPRESSED
+        bne done
+
```

Comment:
Even though the tracker has the order switched around, for bhop purposes this SUPPRESSED check needs to be the very first thing in the function. We should never touch the registers while the channel is suppressed, for any reason.

Reply:
upon closer study of adjacent code, i realize that `#CHANNEL_SUPPRESSED` more closely correlates with m_bGate in Channels2A03.cpp, and not `mEnabled`. upon even closer inspection of the member variable names, i think `mEnabled` is hacked in for FT's implementation of the retrigger command.